Tests/tests/test_SystemAudioPlayer.cpp

```diff
@@ -0,0 +1,1356 @@
+
+        EXPECT_EQ(response, _T("{\"success\":false}"));
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+
+/**
+ * @name  : SAPSetMixerLevelsWithoutOpen
+ * @brief : SettingMixerlevels for invalidPlayer
+ * 
+ * @param[in]   :  id , primaryVolume , player volume
+ * @return      :  {success: false}
+ */
+
+TEST_F(SAPInitializedTest,SAPSetMixerLevelsWithoutOpen) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setMixerLevels"), _T("{\"id\": 1 , \"primaryVolume\": \"180\" ,\"playerVolume\": \"7\" }"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+}
+
+/*******************************************************************************************************************
+ * Test function for SetSmartVolControl
+ * SetSmartVolControl                 :
+ *                Sets the smart volume audio control on the specified player. 
+ *                The plugin can control the smart volume of the content being played back as well as the primary program audio; 
+ *                thus, an application can duck down the volume on the primary program audio when system audio is played 
+ *                and then restore it back when the system audio playback is complete.
+ *
+ *                @return Response object success status
+ * Use case coverage:
+ *                @Success : 1
+ *                @Failure : 5
+ ********************************************************************************************************************/
+/**
+ * @name  : SAPSetSmartVolControl
+ * @brief : SettingSAPSetSmartVolControl for Player
+ * 
+ * @param[in]   :  id, enable, playerAudioLevelThreshold, playerDetectTimeMs, playerHoldTimeMs, primaryDuckingPercent
+ * @return      :  {success: true}
+ */
+
+TEST_F(SAPInitializedTest, SAPSetSmartVolControl) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"websocket\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setSmartVolControl"), _T("{\"id\": ") + std::to_string(speechId) + _T(",\"enable\": true, \"playerAudioLevelThreshold\": 0.1, \"playerDetectTimeMs\": 200, \"playerHoldTimeMs\": 1000, \"primaryDuckingPercent\": 1 }"), response));
```

Comment:
Could you fold this line into two to improve readability?

Reply:
done